sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusRenewOperator.java

```
@@ -0,0 +1,114 @@
+
+        this.isAutoRenewLock = autoLockRenewal;
+        this.maxAutoLockRenewal = maxAutoLockRenewDuration;
+
+    }
+
+    @Override
+    public void subscribe(CoreSubscriber<? super com.azure.messaging.servicebus.ServiceBusReceivedMessage> actual) {
+        source.subscribe(new CoreSubscriber<ServiceBusReceivedMessage>() {
+
+           @Override
+            public void onSubscribe(Subscription s) {
+                actual.onSubscribe(s);
+            }
+
+            @Override
+            public void onNext(ServiceBusReceivedMessage message) {
+
+                final long sequenceNumber = message.getSequenceNumber();
+                final String lockToken = message.getLockToken();
+                LockRenewalOperation renewLockOperation = null;
+
+                if (isAutoRenewLock) {
+                    if (CoreUtils.isNullOrEmpty(lockToken)) {
+                        throw logger.logExceptionAsError(new IllegalStateException(
+                            "Cannot auto-renew message without a lock token on message. Sequence number: " + sequenceNumber));
+                    } else if (message.getLockedUntil() ==  null) {
+                        throw logger.logExceptionAsError(new IllegalStateException(
+                            "Cannot auto-renew message without a lock token until on message. Sequence number: " + sequenceNumber));
+                    }
+
+                    renewLockOperation = new LockRenewalOperation(message.getLockToken(), maxAutoLockRenewal, false,
+                        onRenewLock, message.getLockedUntil());
+                    messageLockContainer.addOrUpdate(lockToken, OffsetDateTime.now().plus(maxAutoLockRenewal),
+                        renewLockOperation);
+                }
+
+                final AtomicBoolean hasError = new AtomicBoolean();
+                try {
+                    actual.onNext(message);
+                } catch (Exception e) {
+                    hasError.set(true);
```

What is the purpose of `hasError`? Also, we are swallowing the exception here. If `actual.onNext()` throws an exception, should we stop the source flux?